src/inet/IPEndPointBasis.h

```diff
@@ -161,12 +161,12 @@ class DLL_EXPORT IPEndPointBasis : public EndPointBasis
     dispatch_queue_t mDispatchQueue;
     dispatch_semaphore_t mSendSemaphore;
 
-    INET_ERROR Bind(IPAddressType aAddressType, IPAddress aAddress, uint16_t aPort, nw_parameters_t aParameters);
+    INET_ERROR Bind(IPAddressType aAddressType, IPAddress aAddress, uint16_t aPort, nw_parameters_t & aParameters);
```

Comment:
Nit: What is the general opinion on pass by reference? I am used to rules like 'pass by reference is const, if require value changes then pass by pointer'.

The reasoning behind it is that this will 'silently' change existing code from making aParameters not change to change for callers. which may  be unintuitive. Is this parameter really changed? if so, is it worth it to make it a pointer instead and eat up the fallout and change caller sites?

Reply:
The policy I've followed since I started using C++ is to use pointers **only** when the value is optional and may be `NULL` or `nullptr`. Otherwise, references are preferred for R/W (in,out) parameters and constant references for R/O (in) parameters.

The code tends to be more efficient with references and it eliminates any `NULL` or `nullptr` checking if the parameter truly is not optional.

openweave-core is somewhat inconsistent on this but tends to follow this pattern.